### ISSUE TITLE
Segmentation fault for sav input

### ISSUE THREAD

**Issue by qiyubio** [weizhouUMICH/SAIGE#249]
Hi there, 

All my test run using extdata/cmd.sh based on sav input failed and give "Segmentation fault" error caused by 'memory not mapped'.
My current environment is:
R/4.0.0
SAIGE_0.39.4
SPAtest_3.1.2

Would you let me know if this is a bug, or some package I was missing? Thanks a lot.

One example of error:
```
Garbage collection 14 = 8+2+4 (level 2) ...
81.6 Mbytes of cons cells used (59%)
20.2 Mbytes of vectors used (32%)

 *** caught segfault ***
address 0x18, cause 'memory not mapped'

Traceback:
 1: setvcfDosageMatrix(vcfFile, vcfFileIndex, vcfField)
 2: SPAGMMATtest(vcfFile = opt$vcfFile, vcfFileIndex = opt$vcfFileIndex,     vcfField = opt$vcfField, bgenFile = opt$bgenFile, bgenFileIndex = opt$bgenFileIndex,     savFile = opt$savFile, savFileIndex = opt$savFileIndex, idstoExcludeFile = opt$idstoExcludeFile,     idstoIncludeFile = opt$idstoIncludeFile, rangestoExcludeFile = opt$rangestoExcludeFile,     rangestoIncludeFile = opt$rangestoIncludeFile, chrom = opt$chrom,     start = opt$start, end = opt$end, IsDropMissingDosages = opt$IsDropMissingDosages,     sampleFile = opt$sampleFile, GMMATmodelFile = opt$GMMATmodelFile,     varianceRatioFile = opt$varianceRatioFile, SAIGEOutputFile = opt$SAIGEOutputFile,     minMAF = opt$minMAF, minMAC = opt$minMAC, numLinesOutput = opt$numLinesOutput,     IsOutputAFinCaseCtrl = opt$IsOutputAFinCaseCtrl, IsOutputNinCaseCtrl = opt$IsOutputNinCaseCtrl,     condition = opt$condition, maxMAFforGroupTest = opt$maxMAFforGroupTest,     groupFile = opt$groupFile, sparseSigmaFile = opt$sparseSigmaFile,     minInfo = opt$minInfo, cateVarRatioMinMACVecExclude = cateVarRatioMinMACVecExclude,     cateVarRatioMaxMACVecInclude = cateVarRatioMaxMACVecInclude,     IsSingleVarinGroupTest = opt$IsSingleVarinGroupTest, dosageZerodCutoff = opt$dosageZerodCutoff,     IsOutputPvalueNAinGroupTestforBinary = opt$IsOutputPvalueNAinGroupTestforBinary,     IsAccountforCasecontrolImbalanceinGroupTest = opt$IsAccountforCasecontrolImbalanceinGroupTest,     method = opt$method, kernel = opt$kernel, weights.beta.rare = weights.beta.rare,     weights.beta.common = weights.beta.common, weightMAFcutoff = opt$weightMAFcutoff,     r.corr = opt$r.corr, weightsIncludeinGroupFile = opt$weightsIncludeinGroupFile,     weights_for_G2_cond = weights_for_G2_cond, IsOutputBETASEinBurdenTest = opt$IsOutputBETASEinBurdenTest,     SPAcutoff = opt$SPAcutoff, IsOutputHetHomCountsinCaseCtrl = opt$IsOutputHetHomCountsinCaseCtrl,     LOCO = opt$LOCO, sampleFile_male = opt$sampleFile_male, is_rewrite_XnonPAR_forMales = opt$is_rewrite_XnonPAR_forMales,     X_PARregion = opt$X_PARregion)
An irrecoverable exception occurred. R is aborting now ...
cmd.sh: line 530:   521 Segmentation fault      Rscript step2_SPAtests.R --vcfFile=./input/seedNumLow_126001_seedNumHigh_127000_nfam_1000_nindep_0.sav --vcfFileIndex=./input/seedNumLow_126001_seedNumHigh_127000_nfam_1000_nindep_0.sav.s1r --vcfField=DS --chrom=chr1 --minMAF=0 --minMAC=0.5 --maxMAFforGroupTest=0.01 --GMMATmodelFile=./output/example_binary.rda --varianceRatioFile=./output/example_binary_cate_v2.varianceRatio.txt --SAIGEOutputFile=./output/example_binary.SAIGE.gene_conditional_withspecifiedWeights.txt.0.37 --numLinesOutput=1 --groupFile=./input/groupFile_geneBasedtest_withWeights.txt --sparseSigmaFile=./output/example_binary_cate_v2.varianceRatio.txt_relatednessCutoff_0.125.sparseSigma.mtx --IsOutputAFinCaseCtrl=TRUE --IsSingleVarinGroupTest=TRUE --IsOutputPvalueNAinGroupTestforBinary=TRUE --IsAccountforCasecontrolImbalanceinGroupTest=TRUE --weightsIncludeinGroupFile=TRUE --weights_for_G2_cond=3,1 --condition=chr1:32302_A/C,chr1:32304_A/C

```

Bests,
Qi

**Comment by qiyubio** [comment]
Work around to pull the container solve the problem. But only 0.39.4 works as expected. 
0.42 and 0.41 give similar error: 
```
Error in t(G1) %*% matrix(obj$residuals, ncol = 1) :
  Cholmod error 'X and/or Y have wrong dimensions' at file ../MatrixOps/cholmod_sdmult.c, line 90
Calls: SPAGMMATtest ... system.time -> SAIGE_SKAT_withRatioVec -> as.vector -> %*% -> %*%

```
Also seems all containers are based on R/3.5.1, which is different from the instruction R/3.6.1, is it a typo?

**Comment by weizhouUMICH** [comment]
Sorry fo the late reply! We have just released a new version 1.0.0. It has computational efficiency improvements for both Step 1 and Step 2 for single-variant and set-based tests. We have created a new program github page https://github.com/saigegit/SAIGE with the documentation provided https://saigegit.github.io/SAIGE-doc/
The program will be maintained by multiple SAIGE developers there. The docker image has been updated. Please feel free to try the version 1.0.0 and report issues if any.

Thanks!
Wei
